prow/kube/config.go

```diff
@@ -98,10 +101,44 @@ func LoadClusterConfigs(kubeconfig, projectedTokenFile string) (map[string]rest.
 		logrus.WithField("tokenfile", projectedTokenFile).Info("Using projected token file")
 	}
 
-	kubeCfgs, currentContext, err := kubeConfigs(kubeconfig)
-	if err != nil {
-		return nil, fmt.Errorf("kubecfg: %v", err)
+	var candidates []string
+	if kubeconfig != "" {
+		candidates = append(candidates, kubeconfig)
+	}
+	if kubeconfigDir != "" {
+		files, err := ioutil.ReadDir(kubeconfigDir)
+		if err != nil {
+			return nil, fmt.Errorf("kubecfg dir: %v", err)
+		}
+		for _, file := range files {
+			filename := file.Name()
+			if kubeconfigSuffix != "" && !strings.HasSuffix(filename, kubeconfigSuffix) {
+				continue
```

Comment:
Why not just simply:

```go
if strings.HasSuffix(filename, suffix){
  candidates = append(...
```

Any string has an empty string suffix, and there's no value in having a negate guard with continue when there's just one command *after* that guard.

Reply:
Why do we even need a suffix filter?

Reply:
> Why do we even need a suffix filter?

I was thinking about filtering out files liek `build01_build01_reg_auth_value.txt`. I just realized it is not enough. We need to filter out by SA name, or split into small secrets.

My bad i did not think about this before.

I think regex on filename is the most flexible. WDYT?

```
oc get secret -n ci build-farm-credentials -o yaml | yq -r '.data|keys[]'
build01_build01_reg_auth_value.txt
build01_build02_reg_auth_value.txt
build01_ci_reg_auth_value.txt
build01_github_client_id
build01_github_client_secret
build01_vsphere_reg_auth_value.txt
build02_github_client_id
build02_github_client_secret
sa.ci-operator.vsphere.config
sa.config-updater.api.ci.config
sa.config-updater.app.ci.config
sa.config-updater.arm01.config
sa.config-updater.build01.config
sa.config-updater.build02.config
sa.config-updater.hive.config
sa.config-updater.vsphere.config
sa.crier.api.ci.config
sa.crier.app.ci.config
sa.crier.arm01.config
sa.crier.build01.config
sa.crier.build02.config
sa.crier.vsphere.config
sa.deck.api.ci.config
sa.deck.app.ci.config
sa.deck.arm01.config
sa.deck.build01.config
sa.deck.build02.config
sa.deck.vsphere.config
sa.dptp-controller-manager.vsphere.config
sa.hook.api.ci.config
sa.hook.app.ci.config
sa.hook.arm01.config
sa.hook.build01.config
sa.hook.build02.config
sa.hook.vsphere.config
sa.jenkins-operator.api.ci.config
sa.prow-controller-manager.api.ci.config
sa.prow-controller-manager.app.ci.config
sa.prow-controller-manager.arm01.config
sa.prow-controller-manager.build01.config
sa.prow-controller-manager.build02.config
sa.prow-controller-manager.vsphere.config
sa.release-bot.app.ci.config
sa.sinker.api.ci.config
sa.sinker.app.ci.config
sa.sinker.arm01.config
sa.sinker.build01.config
sa.sinker.build02.config
sa.sinker.vsphere.config
vsphere_github_client_id
```